### PR TITLE
[ENG-2383] Add Documentation For Collections

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Document all the endpoints for our collections feature.

# Changes

- add files for each collections endpoint
- fixes minor repeated typos

# Side Effects 

None that I know of.

 # Ticket

https://openscience.atlassian.net/browse/ENG-2383

### CODE REVIEW

File: swagger-spec/collections/collected_metadata_list.yaml [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+            last:
+            prev:
+            next:
+            meta:
+              total: 1
+              per_page: 10
+
+
+post:
+  summary: Add Metadata or Subjects to a Entitiy in a Collection
```
Comment (corbinSanders) [inline swagger-spec/collections/collected_metadata_list.yaml:280]: Still a typo

File: swagger-spec/collections/collected_metadata_subjects_relationships.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# /collections/{collection_id}/collected_metadata/{cgm_id}/relationships/subjects/:
+get:
+  summary: Retrieve subjuct metadata for a specific piece of metadata in a collection
```
Comment (corbinSanders) [inline swagger-spec/collections/collected_metadata_subjects_relationships.yaml:3]: Typo

**Review by corbinSanders — CHANGES_REQUESTED** [review]
A few typos. Otherwise looks good

**Comment by corbinSanders** [comment]
LGTM

**Review by jwalz — APPROVED** [review]

**Pull request merged**